onnxruntime/test/testdata/transform/convert_qdq_ops_to_ms_domain.py

```diff
@@ -1,59 +1,153 @@
+Optionally updates zero-points to 16bit data types.
+
 This is used to create models for testing QDQ transformations with the contrib QDQ ops.
 
-Usage: python3 convert_qdq_ops_to_ms_domain.py <onnx model>
+Usage:
+python3 convert_qdq_ops_to_ms_domain.py --input_model <input onnx model> --output_model <output model> --use_16bit_qdq
 
 Models created with this script:
 - qdq_with_multi_consumer_dq_nodes.fixed.qdq_contrib.onnx
+- qdq_with_multi_consumer_dq_nodes.fixed.qdq16_contrib.onnx
 - fusion/constant_folding_dequantizelinear.qdq_contrib.onnx
+- fusion/constant_folding_dequantizelinear.qdq16_contrib.onnx
 - fusion/constant_folding_qdq_node_unit.qdq_contrib.onnx
+- fusion/constant_folding_qdq_node_unit.qdq16_contrib.onnx
 - fusion/constant_folding_qdq_node_unit.graph_output.qdq_contrib.onnx
+- fusion/constant_folding_qdq_node_unit.graph_output.qdq16_contrib.onnx
 """
+import argparse
 import os
+import struct
 import sys
+from typing import Dict
 
 import onnx
+from onnx import shape_inference
 
 QDQ_OPS = ("QuantizeLinear", "DequantizeLinear")
-
-
-def print_usage(prog_name: str):
-    """
-    Prints the program's command-line arguments and usage.
-    """
-
-    print(f"Usage: {prog_name} <onnx model>")
-
-
-def update_qdq_node_domains(graph):
-    """
-    Updates the domain of all QuantizeLinear and DequantizeLinear nodes
-    in a graph to 'com.microsoft'.
-    """
+QDQ_CONVERT_TYPES = {onnx.TensorProto.UINT8: onnx.TensorProto.UINT16, onnx.TensorProto.INT8: onnx.TensorProto.INT16}
+TYPE_TO_STRUCT_LABEL = {
+    onnx.TensorProto.UINT8: "B",
+    onnx.TensorProto.INT8: "b",
+    onnx.TensorProto.UINT16: "H",
+    onnx.TensorProto.INT16: "h",
+}
+
+
+def convert_initializer_to_16bits(initializer: onnx.TensorProto, target_type: onnx.TensorProto.DataType):
+    byte_order = ">" if sys.byteorder == "big" else "<"
+    byte_label = TYPE_TO_STRUCT_LABEL[initializer.data_type]
+    short_label = TYPE_TO_STRUCT_LABEL[target_type]
+
+    # Do not support external data
+    if initializer.HasField("data_location") and initializer.data_location == onnx.TensorProto.EXTERNAL:
+        raise Exception("Do not support initializers with external data")
+
+    # Need to convert raw_data bytes to 16-bit values.
+    # NOTE: For tensors that use .int32_data instead of .raw_data, we don't need any special handling
+    # other than updating the data type. This is because the upper 24 bits are already cleared to zero.
+    if initializer.HasField("raw_data"):
+        num_byte_vals = len(initializer.raw_data)
+
+        # Extract 8-bit values as int32s
+        int32_vals = struct.unpack(f"{byte_order}{num_byte_vals}{byte_label}", initializer.raw_data)
+
+        # Repack int32 values as 16-bit values
+        initializer.raw_data = struct.pack(f"{byte_order}{num_byte_vals}{short_label}", *int32_vals)
+
+    initializer.data_type = target_type
+
+
+def convert_qdq_op_to_16bit(
+    name_to_initializer: Dict[str, onnx.TensorProto],
```

Comment:
unfortunately use of `typing.Dict` is deprecated since 3.9.

instead we can use:
```python
from __future__ import annotations
...
def f(name_to_initializer: dict[str, onnx.TensorProto]):
    ...
```

Reply:
Updated